src/main/java/pi2schema/crypto/LocalDecryptor.java

```diff
@@ -4,22 +4,14 @@
 import pi2schema.crypto.providers.DecryptingMaterialsProvider;
 
 import javax.crypto.Cipher;
-import java.security.GeneralSecurityException;
 import java.util.concurrent.CompletableFuture;
 import java.util.function.BiFunction;
 
 public class LocalDecryptor implements Decryptor {
 
     private final DecryptingMaterialsProvider provider;
 
-    private final BiFunction<Cipher, byte[], CompletableFuture<byte[]>> decrypt =
-            (Cipher cipher, byte[] bytes) -> CompletableFuture.supplyAsync(() -> {
-                try {
-                    return cipher.doFinal(bytes);
-                } catch (GeneralSecurityException e) {
-                    throw new RuntimeException(e);
-                }
-            });
+    private final BiFunction<Cipher, byte[], CompletableFuture<byte[]>> decrypt = CipherSupplier.EXECUTOR;
```

Comment:
we could  even make the supplier return of a crypto function, so we not even have this reference and even removing the knowledge of cipher and encryption from here, so that the encrypt encapsulates the cipher.

However i believe this part will still change quite a bit, so this is already a good step ;)